feature_engine/selection/drop_psi_features.py

```diff
@@ -0,0 +1,566 @@
+        Dataframe containing the PSI values for all features considered.
+
+    Methods
+    -------
+    fit:
+        Find features with high PSI values.
+    transform:
+        Remove features with high PSI values.
+    fit_transform:
+        Fit to the data. Then transform it.
+    """
+
+    def __init__(
+        self,
+        split_col: str = None,
+        split_frac: float = 0.5,
+        split_distinct_value: bool = False,
+        variables: Variables = None,
+        missing_values: str = "ignore",
+        switch: bool = False,
+        threshold: float = 0.25,
+        bins: int = 10,
+        strategy: str = "equal_frequency",
+        min_pct_empty_buckets: float = 0.0001,
+    ):
+        self._check_init_values(
+            split_col,
+            split_frac,
+            split_distinct_value,
+            variables,
+            missing_values,
+            switch,
+            threshold,
+            bins,
+            strategy,
+            min_pct_empty_buckets,
+        )
+        # Set all arguments (except self and variables as attributes)
+        for name, value in vars().items():
+            if name not in ("self", "variables"):
+                setattr(self, name, value)
+
+        # Check the input is in the correct format.
+        self.variables = _check_input_parameter_variables(variables)
+
+    def fit(self, X: pd.DataFrame, y: pd.Series = None):
+        """
+        Assess the features that needs to be dropped because of high PSI values.
+
+        Parameters
+        ----------
+        X : pandas dataframe of shape = [n_samples, n_features]
+            The training dataset.
+
+        y : pandas series. Default = None
+            y is not needed in this transformer. You can pass y or None.
+
+        Returns
+        -------
+        self
+        """
+        # check input dataframe
+        X = _is_dataframe(X)
+
+        # find all variables or check those entered are present in the dataframe
+        self.variables_ = _find_or_check_numerical_variables(X, self.variables)
+
+        # Remove the split columns from the variables list. It is automatically added to
+        # it in variables is not defined at initiation.
+        if self.split_col in self.variables_:
+            self.variables_.remove(self.split_col)
+
+        if self.missing_values == "raise":
+            # check if dataset contains na
+            _check_contains_na(X, self.variables_)
+            _check_contains_inf(X, self.variables_)
+
+        # Split the dataframe into a basis and a measurement dataframe.
+        basis_df, measurement_df = self._split_dataframe(X)
+
+        # Switch base and measurement dataframe if required.
+        if self.switch:
+            measurement_df, basis_df = basis_df, measurement_df
+
+        # Compute the PSI
+        self.psi_ = self._compute_PSI(basis_df, measurement_df, self.bucketer)
+
+        # Select features below the threshold
+        self.features_to_drop_ = self.psi_[
```

Comment:
we should probably make this part of the loop I suggest in my previous comment. Not to loop again.